providers.tf

```diff
@@ -0,0 +1,47 @@
+terraform {
+  required_version = ">= 0.12"
+}
+
+provider "aws" {
+  region = var.aws_region
+}
+
+data "aws_availability_zones" "available" {}
+
+# Not required: currently used in conjunction with using
+# icanhazip.com to determine local workstation external IP
+# to open EC2 Security Group access to the Kubernetes cluster.
+# See workstation-external-ip.tf for additional information.
+provider "http" {}
```

Comment:
You don't have to specify the provider if it doesn't have either a version constraint or some setting.  Terraform will infer this when you run `terraform init` and it sees an `http` resource block.